src/components/App/App.js

```diff
@@ -1,23 +1,75 @@
 import './App.css';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { Routes, Route, useHistory } from 'react-router-dom';
+import { CurrentUserContext } from '../../contexts/CurrentUserContext';
+import mainApi from '../../utils/MainApi';
+import getArticles from '../../utils/NewsApi';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
 import Main from '../Main/Main';
 import SavedNews from '../SavedNews/SavedNews';
 import Footer from '../Footer/Footer';
-import SigningPopup from '../SigningPopup/SigningPopup';
-import SignupPopup from '../SignupPopup/SignupPopup';
+import Login from '../Login/Login';
+import Register from '../Register/Register';
 import Popup from '../Popup/Popup';
+import RegisteredSuccessPopup from '../RegisteredSuccessPopup/RegisteredSuccessPopup';
 import Header from '../Header/Header';
 
 function App() {
+  const [currentUser, setCurrentUser] = useState({
+    name: '',
+    email: '',
+    password: '',
+    _id: ''
+  });
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [receivedArticles, setReceivedArticles] = useState([]);
+  const [savedArticles, setSavedArticles] = useState([]);
+  const [searchKeyword, setSearchKeyword] = useState('');
+  const [isCardsSectionVisible, setIsCardsSectionVisible] = useState(false);
+  const [isLoadingCards, setIsLoadingCards] = useState(false);
+  const [isFoundCards, setIsFoundCards] = useState(false);
+  const [isErrorSearchOccur, setIsErrorSearchOccur] = useState(false);
   const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
   const [isSigningPopupOpen, setIsSigningPopupOpen] = useState(false);
   const [isSignupPopupOpen, setIsSignupPopupOpen] = useState(false);
+  const [isRegisteredSuccessPopupOpen, setIsRegisteredSuccessPopupOpen] = useState(false);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [name, setName] = useState('');
+  const [globalError, setGlobalError] = useState('');
+
+  useEffect(() => {
+    (async function() {
+      if (localStorage.jwt) {
+        mainApi.setHeader(localStorage.jwt);
+        setIsLoggedIn(true);
+        const userInfo = await mainApi.getUserInfo();
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**